sncosmo/models.py

```diff
@@ -2030,3 +2030,95 @@ def propagate(self, wave, flux, phase=None):
+        L0, F0, F1, dL = self._parameters
+        lam_nodes = np.arange(self._minwave, self._maxwave, dL)
+        if lam_nodes.max() < self._maxwave:
+            lam_nodes = np.append(lam_nodes, self._maxwave)
+        siglam_values = self._colordisp(lam_nodes) 
+        
+        siglam_values[lam_nodes < L0] *= 1 + (lam_nodes[lam_nodes < L0] - L0) * F0
+        siglam_values[lam_nodes > L0] *= 1 + (lam_nodes[lam_nodes > L0] - L0) * F1
+        siglam_values *= np.random.normal(size=len(lam_nodes))
+        
+        return lam_nodes, siglam_values
+
+    def propagate(self, wave, flux):
+        """Propagate the effect to the flux."""
+        lam_nodes, siglam_values = self.compute_sigma_nodes()
+        magscat = sine_interp(wave, lam_nodes, siglam_values)
+        return flux * 10**(-0.4 * magscat)
+
+
+class C11(PropagationEffect):
+    """C11 scattering effect for sncosmo.
+    Use COV matrix between the vUBVRI bands from N. Chottard thesis.
+        Implementation is done following arxiv:1209.2482."""
+
+    _param_names = ["C_vU", 'S_f']
+    param_names_latex = ["\rho_\mathrm{vU}", 'S_f']
+    _minwave = 2000
+    _maxwave = 11000
+
+    def __init__(self):
+        """Initialise C11 class."""
+        self._parameters = np.array([0., 1.3])
+
+        # vUBVRI lambda eff
+        self._lam_nodes = np.array([2500.0, 3560.0, 4390.0, 5490.0, 6545.0, 8045.0])
+        
+        # vUBVRI correlation matrix extract from SNANA, came from N.Chotard thesis
+        self._corr_matrix = np.array(
+            [
+                [+1.000000,  0.000000,  0.000000,  0.000000,  0.000000,  0.000000],
+                [ 0.000000, +1.000000, -0.118516, -0.768635, -0.908202, -0.219447],
+                [ 0.000000, -0.118516, +1.000000, +0.570333, -0.238470, -0.888611],
+                [ 0.000000, -0.768635, +0.570333, +1.000000, +0.530320, -0.399538],
+                [ 0.000000, -0.908202, -0.238470, +0.530320, +1.000000, +0.490134],
+                [ 0.000000, -0.219447, -0.888611, -0.399538, +0.490134, +1.000000]
+            ]
+            ) 
+
+        self._corr_matrix[0, 1:] = self._parameters[0] * self._corr_matrix[1, 1:]
+        self._corr_matrix[1:, 0] = self._parameters[0] * self._corr_matrix[1:, 1]
+
+        # vUBVRI sigma
+        self._siglam_values = np.array([0.5900, 0.06001, 0.040034, 0.050014, 0.040017, 0.080007])
+        
+        # Convert corr to cov
+        self._cov_matrix = self._corr_matrix * np.outer(self._siglam_values, 
+                                                        self._siglam_values) 
+        # Rescale covariance as in arXiv:1209.2482
+        self._cov_matrix *= self._parameters[1]
+    
+    def propagate(self, wave, flux):
+        """Propagate the effect to the flux."""
+        siglam_values = np.random.multivariate_normal(np.zeros(len(self._lam_nodes)), self._cov_matrix)
+        
+        inf_mask = wave <= self._lam_nodes[0]
+        sup_mask = wave >= self._lam_nodes[-1]
+        
```

Comment:
I think these first three statements could be moved to `__init__`. Am I missing something?